app/classifier/tasks/dropdown/index.jsx

```diff
@@ -0,0 +1,292 @@
+        optionsKeys[id] = answer.value;
+      } else if (answer.value === null) {
+        optionsKeys[id] = null;
+      } else {
+        optionsKeys[id] = `${optionsKeys[condition]};${answer.value}`;
+      }
+    });
+    return optionsKeys;
+  };
+
+  const getOptionLabel = (value, i) => {
+    const select = props.task.selects[i];
+    const optionsKeys = getOptionsKeys();
+    const optionsKey = select.condition ? optionsKeys[select.condition] : '*';
+    const options = select.options[optionsKey];
+    const optionSelected = options.filter(option => option.value === value);
+    return optionSelected[0].label;
+  };
+
+  return (
+    <div className="classification-task-summary">
+      <div className="question">{props.task.instruction}</div>
+      <div className="answers">
+        {props.annotation.value.map((answer, i) => {
+          let displayAnswer = '';
+          if (answer.option) {
+            displayAnswer = getOptionLabel(answer.value, i);
+          } else {
+            displayAnswer = answer.value;
+          }
+          return (<div key={i} className="answer">
+            <i className="fa fa-arrow-circle-o-right" /> {props.task.selects[i].title} - {displayAnswer}
+          </div>);
+        })}
+      </div>
+    </div>);
+};
+
+DropdownSummary.propTypes = {
+  task: PropTypes.shape({
+    instruction: PropTypes.string,
+    selects: PropTypes.array
+  }),
+  annotation: PropTypes.shape({
+    value: PropTypes.array
+  })
+};
+
+DropdownSummary.defaultProps = {
+  task: {},
+  annotation: {}
+};
+
+export default class DropdownTask extends React.Component {
+  componentWillMount() {
+    this.menus = [];
+  }
+
+  componentDidMount() {
+    this.setDefaultValues();
+
+    if (this.props.autoFocus === true) {
+      this.handleFocus();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.task !== this.props.task) {
+      this.setDefaultValues();
+      if (this.props.autoFocus === true) {
+        this.handleFocus();
+      }
+    }
+  }
+
+  onChangeSelect(i, option) {
+    let newOption = option;
+    if (!newOption) {
+      newOption = { label: null, value: null };
+    }
+    const { value } = this.props.annotation;
+    const options = this.getOptions(i);
+    const newIndex = options.indexOf(newOption);
+
+    if (newIndex === -1) {
+      value[i] = { value: newOption.value, option: false };
+    } else {
+      value[i] = { value: newOption.value, option: true };
+    }
+
+    this.clearRelated(i);
+
+    const newAnnotation = Object.assign({}, this.props.annotation, { value });
+    this.props.onChange(newAnnotation);
+  }
+
+  setDefaultValues() {
+    if (!this.props.annotation.value.length) {
+      const value = this.props.task.selects.map(() => ({ option: false, value: null }));
+      const newAnnotation = Object.assign({}, this.props.annotation, { value });
+      this.props.onChange(newAnnotation);
+    }
+  }
+
+  getOptionsKey(i, key) {
+    const { selects } = this.props.task;
```

Comment:
This is another function that's hard to follow on first reading. I remember this one being hard to get right when working on this component before.